src/components/ConnectButton/ConnectButton.tsx

```diff
@@ -5,24 +5,23 @@ import { useEthereum } from '@/contexts/EthereumContext';
  * Ethereum-based blockchain. It leverages the useEthereum hook
  * from the EthereumContext for managing blockchain connections.
  */
-const ConnectButton: React.FC = () => {
+const ConnectButton: React.FC<ButtonNames> = ({ textBefore, textAfter }) => {
```

Comment:
This is the beginning of a spaghetti code coming 🍝

Each component should own its props, the props of this component should not be in a shared file reused everywhere else. You create spaghetti by doing this.

You can reuse interfaces within the props interface by don't directly connect.

Also it doesn't make sense to have "ButtonNames" it should be at best "ConnectButtonProps" but "Props" is enough


```suggestion
interface Props {
 buttons: xxx,
}

const ConnectButton: React.FC<Props> = ({ ... }) => {
```

See how it is done here:
https://github.com/lukso-network/tools-erc725-inspect/blob/main/components/DataKeysTable/DataKeysTable.tsx#L14

But this is weird imo - you don't need this global shared interface, just use "textBefore" and textAfter - but what does this even mean?

Just use "text" and that's it.

The parent component decides what is textBefore and textAfter and just update text